docs/language/functions/fill.md

```diff
@@ -16,8 +16,18 @@ present in the output type `t` but not in the input.
 Filled fields are added with a `null` value.  Filling is useful when
 you want to be sure that all fields in a schema are present in a record.
 
+The order of newly added fields relative to fields already present is
+undefined.  If maintaining relative order is important, consider applying the
+[`order`](./order.md) or [`shape`](./shape.md) function.
```

Comment:
Fields are added in the order they appear in the input record.

Reply:
After seeing @nwt's comments here we had a quick offline chat and agreed on some text to clarify where `fill` adds the fields rather than the "undefined" I had before. I've pushed a commit with that change.